2020 07 17 pr/add whitelist module control

patriciorangles opened the pull request.

patriciorangles committed:
[IMP][modelu_installation_control]:

Se permite lista blanca de modulos que son permitidos sin contraseña

Commit:
[IMP][module_installation_control]:

COntrolado desinstalacion de modulos de whitelist

pepetreshere commented on module_installation_control/models/module.py:
```diff
@@ -30,7 +30,32 @@ def check_control_modification(self):
+            'ecua_mrp',
+            'trescloud_product_history',
+            'ecua_fixed_assets',
+            'ecua_fixed_assets_ifrs',
```
este no va en el whiteliste, remover ecua_fixed_assets_ifrs

patriciorangles replied:
Ya fue removido!

pepetreshere commented on module_installation_control/models/module.py:
```diff
@@ -30,7 +30,32 @@ def check_control_modification(self):
+            # Esta seccion funciona en la instalacion
+            # Activado Whitelist
+            return False
+        elif not self.name and self._context.get('active_id',False):
+            # Esta seccion funciona en la desinstalacion
+            # verifico contra el whitelist pero por id
+            module_browse = self.env['ir.module.module'].sudo().browse(self._context.get('active_id',False))
```
no permitamos la desinstalación... se van a borrar datos de la bdd, puede ser terrible

patriciorangles replied:
Segun lo conversado se quedo en que se permitia desinstalar para poder eliminar el modulo ribbon, en todo caso voy a controlar que ribbon sea el unico desinstalable sin password

Commit:
[FIX][module_installation_control]:

- Se corrige un modulo que no deberia estar en whitelist
- Se controla la desinstalacion unicamente de web_ribbon, el resto de
modulos se controla por password
